### PR TITLE
Fix/csvexport

### PR DESCRIPTION
**Pull request by martinfalk**
Various fixes and improvements for CSV export.

### CODE REVIEW

File: include/inviwo/core/util/fileextension.h [hunk 1]
```diff
@@ -44,56 +44,68 @@ class IVW_CORE_API FileExtension : public Serializable {
...
+     *
      * @param str Input string  formed like "Text files (*.txt)".
      * @return FileExtension object created from the information given in the input string.
      */
-    static FileExtension createFileExtensionFromString(const std::string &str);
+    static FileExtension createFileExtensionFromString(const std::string& str);
     std::string toString() const;
 
     bool empty() const;
+    bool matchesAll() const;
```
Comment (ResearchDaniel) [inline include/inviwo/core/util/fileextension.h:64]: What does this one do? Same as *? Could add a comment

File: include/inviwo/core/util/fileextension.h [hunk 2]
```diff
@@ -44,56 +44,68 @@ class IVW_CORE_API FileExtension : public Serializable {
...
+     *
+     * @param str   string to be tested
+     * @return true if the string is matched by the FileExtension
+     */
+    bool matches(const std::string& str) const;
 
     virtual void serialize(Serializer& s) const override;
     virtual void deserialize(Deserializer& d) override;
 
     static FileExtension all();
```
Comment (ResearchDaniel) [inline include/inviwo/core/util/fileextension.h:79]: Not part of this PR but any would probably make more sense :)

File: modules/plotting/processors/dataframeexporter.cpp [hunk 3]
```diff
@@ -94,12 +100,16 @@ void DataFrameExporter::exportNow() {
         LogWarn("File already exists: " << exportFile_);
         return;
     }
-    if (exportFile_.getSelectedExtension() == exportFile_.getNameFilters()[0]) {
-        exportAsCSV(separateVectorTypesIntoColumns_);
-    } else if (exportFile_.getSelectedExtension() == exportFile_.getNameFilters()[1]) {
+    if (exportFile_.getSelectedExtension() == xmlExtension_) {
```
Comment (ResearchDaniel) [inline modules/plotting/processors/dataframeexporter.cpp:103]: What if the file dialog has not been used, i.e. set via scripting. Should it be:
`exportFile_.getSelectedExtension() == xmlExtension_ || filesystem::getFileExtension(exportFile_) == xmlExtension_`

File: src/core/util/fileextension.cpp [hunk 4]
```diff
@@ -88,6 +88,19 @@ std::string FileExtension::toString() const {
 
 bool FileExtension::empty() const { return extension_.empty() && description_.empty(); }
 
+bool FileExtension::matchesAll() const { return extension_ == "*"; }
+
+bool FileExtension::matches(const std::string& str) const {
+    if (empty()) {
+        return false;
+    }
+    if (extension_.empty()) {
```
Comment (ResearchDaniel) [inline src/core/util/fileextension.cpp:97]: or matchesAll()?

**Review by ResearchDaniel — CHANGES_REQUESTED** [review]
Might be some errors. 
Consider renaming FileExtension::all() to FileExtension::any()  (and in that case matchesAny)

File: include/inviwo/core/util/fileextension.h (hunk 2 above)
Comment (petersteneteg) [reply include/inviwo/core/util/fileextension.h:79]: the filter is usually called 
>   "*", "All Files"

not "Any Files" hence I think ```all``` makes more sense...

File: include/inviwo/core/util/fileextension.h (hunk 1 above)
Comment (martinfalk) [reply include/inviwo/core/util/fileextension.h:64]: this one returns true, if the extension only contains an asterisk, i.e. the `FileExtension` matches all files. 
```c++
bool FileExtension::matchesAll() const { return extension_ == "*"; }
```
Will add a comment.

File: include/inviwo/core/util/fileextension.h (hunk 2 above)
Comment (martinfalk) [reply include/inviwo/core/util/fileextension.h:79]: agree with Peter. I had a look at various file dialogs in different applications.

File: modules/plotting/processors/dataframeexporter.cpp (hunk 3 above)
Comment (martinfalk) [reply modules/plotting/processors/dataframeexporter.cpp:103]: Good point. I will move the extension checking from the widget into the fileproperty itself.

File: src/core/util/fileextension.cpp (hunk 4 above)
Comment (martinfalk) [reply src/core/util/fileextension.cpp:97]: not necessarily, someone, e.g. @ResearchDaniel, might come along and decide to add a custom `FileExtension("*", "Matches any file")` to the property. Then the description of the FileExtension would not match `::all()`

**Review by ResearchDaniel — APPROVED** [review]

**Pull request closed without merging**